src/main/scala/analysis/Local.scala

```diff
@@ -0,0 +1,351 @@
+             visitPointerArithmeticOperation(n, lhsCell, arg, 0)
+
+            case MemoryLoad(mem, index, endian, size) => // Rx = Mem[Ry], merge Rx and pointee of Ry (E(Ry))
+              val byteSize = (size.toDouble/8).ceil.toInt
+              lhsCell.node.get.flags.read  = true
+              global = isGlobal(index, n, byteSize)
+              stack = isStack(index, n)
+              if global.isDefined then
+                graph.mergeCells(lhsCell,graph.adjust(graph.find(global.get).getPointee))
+              else if stack.isDefined then
+                graph.mergeCells(lhsCell, graph.adjust(graph.find(stack.get).getPointee))
+              else
+                index match
+                  case BinaryExpr(op, arg1: Variable, arg2) =>
+                    evaluateExpression(arg2, constProp(n)) match
+                      case Some(v) =>
+//                        assert(varToSym(n).contains(arg1))
+                        val offset = v.value
+                        visitPointerArithmeticOperation(n, lhsCell, arg1, byteSize, true, offset)
+                      case None =>
+//                        assert(varToSym(n).contains(arg1))
+                        visitPointerArithmeticOperation(n, lhsCell, arg1, byteSize, true, 0, true)
+                  case arg: Variable =>
+//                    assert(varToSym(n).contains(arg))
+                    visitPointerArithmeticOperation(n, lhsCell, arg, byteSize, true)
+                  case _ => ???
+            case _ =>
+              unsupportedPointerArithmeticOperation(n, expr, lhsCell)
+
+      case MemoryAssign(memory,  ind: Expr, expr: Expr, endian: Endian, size: Int, label) if unwrapPaddingAndSlicing(expr).isInstanceOf[Variable] => // if value is a literal ignore it
+        // Mem[Ry] = Rx
+        val value: Variable = unwrapPaddingAndSlicing(expr).asInstanceOf[Variable]
+        val index: Expr = unwrapPaddingAndSlicing(ind)
+        reachingDefs(n)(value).foreach(visit)
+        val byteSize = (size.toDouble/8).ceil.toInt
+        val global = isGlobal(index, n, byteSize)
+        val stack = isStack(index, n)
+        val addressPointee: DSC =
+          if global.isDefined then
+            graph.adjust(graph.find(global.get).getPointee)
+          else if stack.isDefined then
+            graph.adjust(graph.find(stack.get).getPointee)
+          else
+            index match
+              case BinaryExpr(op, arg1: Variable, arg2) =>
+                evaluateExpression(arg2, constProp(n)) match
+                  case Some(v) =>
+//                    assert(varToSym(n).contains(arg1))
+                    val offset = v.value
+                    visitPointerArithmeticOperation(n, DSN(Some(graph)).cells(0), arg1, byteSize, true, offset)
+                  case None =>
+//                    assert(varToSym(n).contains(arg1))
+                    visitPointerArithmeticOperation(n, DSN(Some(graph)).cells(0), arg1, byteSize, true, 0, true)
```

Comment:
This is assuming that the operation is addition and that the value of arg2 is 0 if it could not be evaluated.

Reply:
yes, this is should be more clear but in this case the resulting node is collapsed with the second true passed to the function, therefore, the offset doesn't matter

Reply:
Is it the case that any other operation is equivalent to Rx = Ry + 0 though? The result isn't even necessarily offset from Ry if it's not addition or subtraction?

Reply:
Shouldn't it be unsupportedPointerArithmeticOperation that's used for this case (where the operation isn't addition) not visitPointerArithmeticOperation?

Reply:
yes we can only handle addition precisely, I added a guard to the binary expression case to check for add op. otherwise unsupportedPointerArithmetic should be called